pkg/controller/pravegacluster/pravegacluster_controller.go

```diff
@@ -301,9 +308,13 @@ func (r *ReconcilePravegaCluster) syncClusterSize(p *pravegav1alpha1.PravegaClus
 		return err
 	}
 
-	err = r.syncSegmentStoreSize(p)
-	if err != nil {
-		return err
+	/*this condition is to stop syncSegmentstore version from running when we are updating the segment store version as in case of
+	updapting the ss from version below 06 to version above 07 get() call in syncSegmentstore will result in error */
+	if p.Status.CurrentVersion == p.Spec.Version {
```

Comment:
This is needed to avoid client.get() inside the syncSegmentstore version() as that will result in error while upgrading to a version above 07 as it will get the lastest name of the ss which is not yet created as the creation code is inside the upgrade.